test/extended/images/mirror.go

```diff
@@ -0,0 +1,367 @@
+	if len(m) == 0 {
+		err = fmt.Errorf("unexpected status code (expected %s): %s", statusCode, out)
+	}
+	pod.NotErr(err)
+}
+
+func testNewBuild(oc *exutil.CLI) (string, string) {
+	isName := "mirror-" + getRandName()
+	istName := isName + ":latest"
+
+	testDockerfile := fmt.Sprintf(`FROM busybox:latest
+RUN echo %s > /1
+RUN echo %s > /2
+RUN echo %s > /3
+`,
+		getRandName(),
+		getRandName(),
+		getRandName(),
+	)
+
+	err := oc.Run("new-build").Args("-D", "-", "--to", istName).InputString(testDockerfile).Execute()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.By("starting a test build")
+	bc, err := oc.BuildClient().Build().BuildConfigs(oc.Namespace()).Get(isName, metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(*bc.Spec.Source.Dockerfile).To(o.Equal(testDockerfile))
+
+	g.By("expecting the Dockerfile build is in Complete phase")
+	err = exutil.WaitForABuild(oc.BuildClient().Build().Builds(oc.Namespace()), isName+"-1", nil, nil, nil)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.By(fmt.Sprintf("checking for the imported tag: %s", istName))
+	ist, err := oc.ImageClient().Image().ImageStreamTags(oc.Namespace()).Get(istName, metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return isName, ist.Image.Name
+}
+
+var _ = g.Describe("[Feature:ImageMirror][registry][Slow] Image mirror", func() {
+	defer g.GinkgoRecover()
+
+	var oc = exutil.NewCLI("image-mirror", exutil.KubeConfigPath())
+
+	g.It("mirror image from integrated registry to external registry", func() {
+		g.By("get user credentials")
+		user, token := getCreds(oc)
+
+		g.By("fetch the URL of integrated registry server")
+		registryHost, err := GetDockerRegistryURL(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Set up and fetch the URL of external registry server")
+		pod := NewTestPod(oc, "oc-image-mirror-test-1", testPod1).Run()
+
+		g.By("get the protocol of integrated registry server")
+		schema, err := getRegistrySchema(pod, registryHost)
+		pod.NotErr(err)
+
+		framework.Logf("the protocol is %s://%s", schema, registryHost)
+
+		g.By("generate config.json for integrated registry server")
+		genDockerConfig(pod, registryHost, user, token)
+
+		g.By("calling oc new-build with Dockerfile")
+		isName, imgName := testNewBuild(oc)
+
+		repoName := oc.Namespace() + "/" + isName
+
+		g.By("Check that we have it in the integrated registry server")
+		requestHasStatusCode(pod, fmt.Sprintf("%s://%s/v2/%s/manifests/%s", schema, registryHost, repoName, imgName), token, "200")
+
+		g.By("Check that we do not have it in the external registry server")
+		requestHasStatusCode(pod, fmt.Sprintf("http://127.0.0.1:5001/v2/%s/manifests/%s", repoName, imgName), "", "404")
+
+		g.By("Mirror image from the integrated registry server to the external registry server")
+		command := fmt.Sprintf("cd /tmp; oc image mirror %s/%s:latest %s/%s:stable --insecure=true",
+			registryHost, repoName, "127.0.0.1:5001", repoName,
```

Comment:
you're running this registry in a pod, how is it available on 127.0.0.1?

shouldn't you be accessing the pod ip, or creating a service and accessing the service ip?

Reply:
According to [documentation](https://kubernetes.io/docs/concepts/workloads/pods/pod):
> Containers within a pod share an IP address and port space, and can find each other via localhost

Reply:
sorry, I missed the fact that you're exec'ing this command inside the registry pod.